core/src/test/java/com/ericsson/bss/cassandra/ecchronos/core/repair/TestOnDemandStatus.java

```diff
@@ -0,0 +1,305 @@
+        assertThat(row.getString(STATUS_COLUMN_NAME)).isEqualTo("failed");
+    }
+
+    @Test
+    public void testGetOngoingJobsNoJobs()
+    {
+        OnDemandStatus onDemandStatus = new OnDemandStatus(getNativeConnectionProvider());
+
+        Set<OngoingJob> ongoingJobs = onDemandStatus.getMyOngoingJobs(myReplicationState);
+
+        assertThat(ongoingJobs).isEmpty();
+    }
+
+    @Test
+    public void testGetOngoingJobsWithNewJob()
+    {
+        OnDemandStatus onDemandStatus = new OnDemandStatus(getNativeConnectionProvider());
+
+        UUID jobId = UUID.randomUUID();
+        int hashValue = 1;
+        TableReference tableReference = myTableReferenceFactory.forTable(KEYSPACE_NAME, TABLE_NAME);
+        Map<LongTokenRange, ImmutableSet<Node>> tokenMap = new HashMap<>();
+        when(myReplicationState.getTokenRangeToReplicas(tableReference)).thenReturn(tokenMap );
+
+        onDemandStatus.addNewJob(jobId, tableReference, hashValue);
+
+        Set<OngoingJob> ongoingJobs = onDemandStatus.getMyOngoingJobs(myReplicationState);
+
+        assertThat(ongoingJobs.size()).isEqualTo(1);
+
+        OngoingJob ongoingJob = ongoingJobs.iterator().next();
+        assertThat(ongoingJob.getJobId()).isEqualTo(jobId);
+        assertThat(ongoingJob.getTableReference().getKeyspace()).isEqualTo(KEYSPACE_NAME);
+        assertThat(ongoingJob.getTableReference().getTable()).isEqualTo(TABLE_NAME);
+        assertThat(ongoingJob.getRepairedTokens()).isEmpty();
+    }
+
+    @Test
+    public void testGetOngoingJobsWithUpdatedJob()
+    {
+        OnDemandStatus onDemandStatus = new OnDemandStatus(getNativeConnectionProvider());
+
+        UUID jobId = UUID.randomUUID();
+        int hashValue = 1;
+        TableReference tableReference = myTableReferenceFactory.forTable(KEYSPACE_NAME, TABLE_NAME);
+        Map<LongTokenRange, ImmutableSet<Node>> tokenMap = new HashMap<>();
+        when(myReplicationState.getTokenRangeToReplicas(tableReference)).thenReturn(tokenMap );
+        onDemandStatus.addNewJob(jobId, tableReference, hashValue);
+
+        Set<LongTokenRange> expectedRepairedTokens = new HashSet<>();
+        expectedRepairedTokens.add(new LongTokenRange(-50L, 700L));
+        Set<UDTValue> repairedTokens = new HashSet<>();
+        repairedTokens.add(onDemandStatus.createUDTTokenRangeValue(-50L, 700L));
+        onDemandStatus.updateJob(jobId, repairedTokens);
+
+        Set<OngoingJob> ongoingJobs = onDemandStatus.getMyOngoingJobs(myReplicationState);
+
+        assertThat(ongoingJobs.size()).isEqualTo(1);
+
+        OngoingJob ongoingJob = ongoingJobs.iterator().next();
+        assertThat(ongoingJob.getJobId()).isEqualTo(jobId);
+        assertThat(ongoingJob.getTableReference().getKeyspace()).isEqualTo(KEYSPACE_NAME);
+        assertThat(ongoingJob.getTableReference().getTable()).isEqualTo(TABLE_NAME);
+        assertThat(ongoingJob.getRepairedTokens()).isEqualTo(expectedRepairedTokens);
+    }
```

Comment:
I think it would be good to add a test case where the table reference uuid does not match (and make sure that we don't pick it up).